android/java/org/chromium/chrome/browser/preferences/BravePreferenceFragment.java

```diff
@@ -0,0 +1,89 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.preferences;
+
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager.NameNotFoundException;
+import android.preference.Preference;
+import android.preference.PreferenceFragment;
+import java.util.HashMap;
+import org.chromium.base.Log;
+import org.chromium.chrome.browser.ChromeVersionInfo;
+import org.chromium.chrome.browser.preferences.PrefServiceBridge.AboutVersionStrings;
+import org.chromium.chrome.browser.preferences.website.SiteSettingsCategory;
+import org.chromium.chrome.browser.preferences.website.SiteSettingsPreferences;
+
+/**
+ * Base class for preferences.
+ */
+public class BravePreferenceFragment extends PreferenceFragment {
+    private static final String PREF_SAFE_BROWSING = "safe_browsing";
+    private static final String PREF_DO_NOT_TRACK = "do_not_track";
+    private static final String PREF_USAGE_AND_CRASH_REPORTING = "usage_and_crash_reports";
+    private static final String PREF_APPLICATION_VERSION = "application_version";
+
+    private static final String VERSION_MASK = "Brave %s, Chromium %s";
+
+    private final HashMap<String, Preference> mRemovedPreferences = new HashMap<>();
+
+    @Override
```

Comment:
it looks like `updatePreferences` is called in `MainPreferences` in other places so just doing this on `onResume` seems fragile because new calls could be added in other places.

Reply:
maybe `PreferenceScreen` should subclass `BravePreferenceGroup` and override `onPrepareAddPreference`? Then we wouldn't need to subclass all the preference fragments because the PreferenceGroup would filter them out of the PreferenceScreen?

Reply:
we do not rely on `updatePreferences`, why do we care whether it's called anywhere else?